### ISSUE TITLE
ValueError: max_workers must be greater than 0

### ISSUE THREAD

**Issue by mikkoc** [cloudreach/sceptre#163]
Just ran into this error when deleting a whole environment.
All the stacks were successfully deleted, the error came up after sceptre was finished.
Not sure it's reproducible...

```
# sceptre --var-file=environments/prod.yaml delete-env base
[...]
[2017-07-10 13:26:07] - sceptre.stack - base/network/vpc - delete complete
Traceback (most recent call last):
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/bin/sceptre", line 11, in <module>
    load_entry_point('sceptre==1.1.1', 'console_scripts', 'sceptre')()
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/cli.py", line 93, in decorated
    return func(*args, **kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/cli.py", line 321, in delete_env
    response = env.delete()
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/environment.py", line 143, in delete
    "delete", threading_events, stack_statuses, delete_dependencies
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/helpers.py", line 126, in decorated
    response = future.result()
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/concurrent/futures/_base.py", line 422, in result
    return self.__get_result()
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/concurrent/futures/thread.py", line 62, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/helpers.py", line 110, in decorated
    return func(self, *args, **kwargs)
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/sceptre/environment.py", line 202, in _build
    with ThreadPoolExecutor(max_workers=num_stacks) as executor:
  File "/Users/mikkocaldara/tfl/git/tfl-eos-infrastructure/venv/lib/python2.7/site-packages/concurrent/futures/thread.py", line 105, in __init__
    raise ValueError("max_workers must be greater than 0")
ValueError: max_workers must be greater than 0
```

**Comment by theseanything** [comment]
Could you provide a tree of the environment your are trying to delete? Also could provide some details about your environment?

**Comment by mikkoc** [comment]
Ok, so there's some empty directories, that's probably causing the issue:

```
tree config/base
config/base
├── audit
├── iam
└── network
    ├── common-sg.yaml
    ├── eips.yaml
    ├── s3-endpoint.yaml
    ├── subnets.yaml
    ├── vpc-flow-logs.yaml
    └── vpc.yaml
```

**Comment by terrybonds** [comment]
Just in case it's useful... 

Having a file with a .yml extension as the sole file in a folder causes this issue as well.  (I assume Sceptre sees it as an empty directory).

**Comment by mikkoc** [comment]
That would make sense since `.yml` is not supported (yet?).
See https://github.com/cloudreach/sceptre/issues/67 for details.

**Comment by theseanything** [comment]
Fixed with #190
